### ISSUE TITLE
Unable to migrate on ubuntu 20.04

### ISSUE THREAD

**Issue by Ovler-Young** [leafac/kill-the-newsletter#108]
OS: Ubuntu 20.04 x86_64

Using version 2.0.1 to migrate:

### log:
```
node:internal/modules/cjs/loader:1602
  return process.dlopen(module, path.toNamespacedPath(filename));
                 ^

Error: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /root/kill-the-newsletter/kill-the-newsletter/_/node_modules/better-sqlite3/build/Release/better_sqlite3.node)
    at Module._extensions..node (node:internal/modules/cjs/loader:1602:18)
    at Module.load (node:internal/modules/cjs/loader:1287:32)
    at Module._load (node:internal/modules/cjs/loader:1103:12)
    at Module.require (node:internal/modules/cjs/loader:1310:19)
    at require (node:internal/modules/helpers:179:18)
    at bindings (/root/kill-the-newsletter/kill-the-newsletter/_/node_modules/bindings/bindings.js:112:48)
    at new Database (/root/kill-the-newsletter/kill-the-newsletter/_/node_modules/better-sqlite3/lib/database.js:48:64)
    at new Database (file:///root/kill-the-newsletter/kill-the-newsletter/_/node_modules/@radically-straightforward/sqlite/build/index.mjs:104:9)
    at file:///root/kill-the-newsletter/kill-the-newsletter/_/build/index.mjs:44:24 {
  code: 'ERR_DLOPEN_FAILED'
}

Node.js v22.1.0
```

The bounded better-sqlite3 actually needs GLIBC_2.33, that is ubuntu 22.04, as it's builded on top of the 22.04 runner of Github. But maybe we can build it on `ubuntu-20.04` in the Github Action, which will fix the problem.

**Comment by Ovler-Young** [comment]
I'm pleased to report that building on Ubuntu 20.04 has successfully resolved the issue!

I made some adjustments to the workflow, as the code in version 2.0.1 doesn't compile with the latest Node.js version (v23). To ensure compatibility, I locked the Node.js version to 22.1.0, which matches the version used in your releases.  You can see my successful build run at: https://github.com/Ovler-Young/kill-the-newsletter/actions/runs/11599220752

With these changes, the build process now works smoothly.

One minor hiccup I encountered was that my previous failed attempt had generated a malformed database. I had to remove this incorrect database to allow the migration to proceed correctly.

**Comment by leafac** [comment]
Hi @Ovler-Young,

Thanks for reaching out.

I’m glad you found a solution to the situation. You’re right in noticing that I tend to support and test on the latest versions. Speaking of which, what went wrong in building with Node.js 23? We better fix that 😅

**Comment by Ovler-Young** [comment]
Building with Node.js 23 failed because the better_sqlite3 failed. Again. 

Error log can be found [here](https://github.com/Ovler-Young/kill-the-newsletter/actions/runs/11579405150/job/32235514984).

After some check I found all runs in Node.js 22.x works fine, so I switched, and it works fine then.

It sees that better_sqlite3 had already fixed this problem in v11.5.0, and is fixed in radically-straightforward in commit https://github.com/radically-straightforward/radically-straightforward/commit/91b7ea0b9da7574059a5f6a8fa42a20a6fdea4ac !

There shouldn't have any problem on Node.js 23 now I believe.
